src/bot/integrations/tipeeestream.ts

```diff
@@ -0,0 +1,182 @@
+  async connect () {
+    this.disconnect();
+
+    if (this.username.trim() === '' || this.apiKey.trim() === '' || !this.enabled) {
+      return;
+    }
+
+    // get current avaliable host and port
+    // destucted response from https://api.tipeeestream.com/v2.0/site/socket
+    // example response: { "code": 200, "message": "success", "datas": { "port": "443", "host": "https://sso-cf.tipeeestream.com" } }
+    const { data: { datas: { host, port }}} = await axios.get('https://api.tipeeestream.com/v2.0/site/socket');
+
+    this.socketToTipeeestream = io(`${host}:${port}`,
+      {
+        reconnection: true,
+        reconnectionDelay: 1000,
+        reconnectionDelayMax: 5000,
+        reconnectionAttempts: Infinity,
+        query: {
+          access_token: this.apiKey.trim(),
+        },
+      });
+
+    if (this.socketToTipeeestream !== null) {
+      this.socketToTipeeestream.on('connect_error', (e) => {
+        error(chalk.red('TIPEEESTREAM.COM:') + ' error while connecting, ' + e);
+      });
+      this.socketToTipeeestream.on('connect', () => {
+        if (this.socketToTipeeestream !== null) {
+          this.socketToTipeeestream.emit('join-room', { room: this.apiKey.trim(), username: this.username.trim() });
+        }
+        info(chalk.yellow('TIPEEESTREAM.COM:') + ' Successfully connected socket to service');
+      });
+      this.socketToTipeeestream.on('reconnect_attempt', () => {
+        info(chalk.yellow('TIPEEESTREAM.COM:') + ' Trying to reconnect to service');
+      });
+      this.socketToTipeeestream.on('disconnect', () => {
+        info(chalk.yellow('TIPEEESTREAM.COM:') + ' Socket disconnected from service');
+        this.disconnect();
+        this.socketToTipeeestream = null;
+      });
+
+      this.socketToTipeeestream.on('new-event', async (data) => {
+        if (data.event.type !== 'donation') {
+          return;
+        }
+        const { amount, message } = data.event.parameters;
+        const username = data.event.parameters.username.toLowerCase();
+        const donationCurrency = data.event.parameters.currency;
+
+        eventlist.add({
+          event: 'tip',
+          amount,
+          currency: donationCurrency,
+          username,
+          message,
+          timestamp: Date.now(),
+        });
+
+        events.fire('tip', {
+          username,
+          amount: parseFloat(amount).toFixed(2),
+          currency: donationCurrency,
+          amountInBotCurrency: Number(currency.exchange(amount, donationCurrency, currency.mainCurrency)).toFixed(2),
+          currencyInBot: currency.mainCurrency,
+          message,
+        });
+
+        alerts.trigger({
+          event: 'tips',
+          name: username,
+          amount: Number(parseFloat(amount).toFixed(2)),
+          currency: donationCurrency,
+          monthsName: '',
+          message,
+          autohost: false,
+        });
+
+        if (!data._is_test_alert) {
+          const getUser = async (username, id) => {
```

Comment:
```suggestion
          const getUser = async (username) => {
```

Reply:
btw: isn't this already an function? If not I would probably move it to users library and name it something like `getUserByUsername`

Reply:
Right. This method is missing.

Reply:
Can you move it to `users` then? would be better to have it in common place.

Reply:
Ofc, does that function code LGTY?

Reply:
Done in https://github.com/sogehige/sogeBot/pull/3334